### PR TITLE
Bug 1228633 - Ignore transient network errors when advancing login st…

### PR DESCRIPTION
**Pull request by ncalexan**
…ate machine. r?rnewman

### CODE REVIEW

File: Account/FxALoginStateMachine.swift [hunk 1]
```diff
@@ -115,6 +115,9 @@ class FxALoginStateMachine {
                                 log.error("Unknown error: \(error.description).  Transitioning to Separated.")
                                 return separated
                             }
+                        case let .Local(localError) where localError.domain == NSURLErrorDomain:
+                            log.warning("Local networking error: \(result.failureValue!).  Assuming transient and not transitioning.")
```
Comment (rnewman) [inline Account/FxALoginStateMachine.swift]: Does this log the code? If not, let's do that too. (And eventually we want to record telemetry for this…!)

File: AccountTests/FxALoginStateMachineTests.swift [hunk 2]
```diff
@@ -130,6 +143,20 @@ class FxALoginStateMachineTests: XCTestCase {
...
     }
 
+    func testAdvanceFromEngagedAfterVerifiedWithoutNetwork() {
+        // Advancing from engaged after verified, but during outage, stays put.
+        withMachine(MockFxALoginClientWithoutNetwork()) { stateMachine in
+            let engagedState = FxAStateTests.stateForLabel(.EngagedAfterVerified)
+
+            let e = self.expectationWithDescription("Wait for login state machine.")
+            stateMachine.advanceFromState(engagedState, now: 0).upon { newState in
+                XCTAssertEqual(newState.label.rawValue, engagedState.label.rawValue)
```
Comment (rnewman) [inline AccountTests/FxALoginStateMachineTests.swift]: A reminder that in most cases — those that don't chain through the main thread — you can do

```
withMachine(…) {
    …
    let newState = stateMachine.advanceFromState(engagedState, now: 0).value.successValue
    XCTAssertEqual(newState.label, engageState.label)
}
```

File: Account/FxALoginStateMachine.swift (hunk 1 above)
Comment (ncalexan) [inline Account/FxALoginStateMachine.swift]: It does: `result.failureValue` is an `FxAClientError`, which is verbose in the case of `.Local`.  (See the existing logging that appears in your bug report.)

File: AccountTests/FxALoginStateMachineTests.swift (hunk 2 above)
Comment (ncalexan) [inline AccountTests/FxALoginStateMachineTests.swift]: Mmm.  I'm not a fan of such implicit threading assumptions, although I guess it tends to be clear why a thing broke.

**Comment by rnewman** [comment]
v1.x: 42e35c7
master: a9fe225

**Pull request closed without merging**